src/Symfony/Component/HttpKernel/Controller/LegacyArgumentResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Responsible for the creation of the action arguments.
+ *
+ * @deprecated This class is deprecated since 3.1 and will be removed in 4.0. Please use the ArgumentResolver::class instead.
```

Comment:
the class is also deprecated (which is why it is named LegacyArgumentResolver btw), as it will be removed in 4.0. Code should use the new system instead